pkg/allocation/converters/converter.go

```diff
@@ -77,7 +88,7 @@ func ConvertGSAToAllocationRequest(in *allocationv1.GameServerAllocation) *pb.Al
 			Enabled: in.Spec.MultiClusterSetting.Enabled,
 		},
 		RequiredGameServerSelector: convertInternalLabelSelectorToLabelSelector(&in.Spec.Required),
-		MetaPatch: &pb.MetaPatch{
+		Metadata: &pb.MetaPatch{
```

Comment:
This wouldn't work for [multi-cluster allocation](https://github.com/googleforgames/agones/blob/d8748dfb82dde85994209b2b8505ef9ee652a818/pkg/gameserverallocations/allocator.go#L350-L353). If one cluster has the field and another one does not have the field yet, it will ignore the `metadata`. In that case, we should do a double write to both `MetaPatch` and `Metadata`. So my earlier comment to return error for double write, wouldn't work. You can instead on the API add a comment that `MetaPatch` is ignored if `Metadata` is set.

Reply:
Oh yeah - that's a good catch!

Reply:
I've added the double write in the GSA to allocation request conversion and updated the tests accordingly.